File: SalonDeEventosBack_End/salon/src/main/java/com/m5a/salon/model/entity/ImgProducto.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.m5a.salon.model.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import java.io.Serializable;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 *
 * @author LaptopSA
 */
@Entity
@Data
@AllArgsConstructor
@Table(name = "ImagenesProductos")
@NoArgsConstructor
public class ImgProducto implements Serializable {

    /**
     *
     */
    private static final long serialVersionUID = 1L;
    /**
     *
     */
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "imgProdId")
    private Long imgProdId;

    @Column(name = "imgProdNombre")
    private String imgProdNombre;

    @Column(name = "imgProdUrl")
    private String imgProdUrl;

    @ManyToOne
    @JoinColumn(name = "prodId", referencedColumnName = "prodId")
    private ProductoServicio prodId;

}

File: SalonDeEventos_FrontEnd/SalonDeEventos/src/app/modelo/imgProducto.ts
import { ProductoServicio } from "./producto-servicio";

export class ImgProducto {
    imgProdId: number;
    imgProdNombre: string;
    imgProdUrl: string;
    prodId:ProductoServicio;
    
    constructor(
        imgProdId?: number,
        imgProdNombre?: string,
        imgProdUrl?: string,
        prodId?: ProductoServicio

    ) {
        this.imgProdId = imgProdId||0;
        this.imgProdNombre = imgProdNombre || "";
        this.imgProdUrl = imgProdUrl || "";
        this.prodId = prodId || new ProductoServicio;
    }

}